### PR TITLE
Add Create new dashboard widget action

### PR DESCRIPTION
**Pull request by maxiadlovskii**
## Motivation and Context
fix: #7629 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by linuspahl** [comment]
For me it was a bit confusing to have two actions for moving the widget to a dashboard.

We could also just have one action "Copy to dashboard" and in the modal we offer the option to copy the widget to a new dashboard. The UI could look similar to the "Move to Page" modal 

image

With the difference that you can only select either the checkbox or a dashboard. What do you think?

**Comment by maxiadlovskii** [comment]
> For me it was a bit confusing to have two actions for moving the widget to a dashboard.
> 
> We could also just have one action "Copy to dashboard" and in the modal we offer the option to copy the widget to a new dashboard. The UI could look similar to the "Move to Page" modal
> 
> image
> 
> With the difference that you can only select either the checkbox or a dashboard. What do you think?

@linuspahl I thought about that... but in this case, the user must do 3 clicks instead of 1.  I think 1 click would be better.

File: graylog2-web-interface/src/views/components/AdaptableQueryTabs.tsx [hunk 1]
```diff
@@ -224,6 +227,18 @@ const _onCopyToDashboard = (selectedDashboardId: string | undefined | null) => a
     .then(addPageToDashboard(targetDashboard, view, queryId));
 };
 
+const _onCreateNewDashboard = async (view: View, queryId: string, history: HistoryFunction) => {
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/AdaptableQueryTabs.tsx:230]: Looks lie we do not need `async` here

File: graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.tsx [hunk 2]
```diff
@@ -16,23 +16,25 @@
...
 import type { SearchParams } from 'stores/PaginationTypes';
 
 type Props = {
   activeDashboardId?: string,
   onCancel: () => void,
-  onSubmit: (selectedDashboardId: string | undefined | null) => Promise<void>,
+  onCopyToDashboard: (selectedDashboardId: string | undefined | null) => Promise<void>,
   submitButtonText: string,
   submitLoadingText: string,
+  onCreateNewDashboard?: () => Promise<void>,
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.tsx:30]: Currently this function does not seem to return a promise.

**Review by linuspahl — APPROVED** [review]
Works well!

File: graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.tsx (hunk 2 above)
Comment (maxiadlovskii) [reply graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.tsx:30]: We need it. in the end, we have part of the code 

```
    submitHandler().then(() => {
      if (isMounted.current) {
        setIsSubmitting(false);
      }
    });
```

where `submitHandler` 

`submitHandler = async () => (createNewDashboard ? onCreateNewDashboard() : onCopyToDashboard(selectedDashboard));`

handle copying widget to dashboard or creating a new one

**Pull request merged**